### PR TITLE
Search input component, simplified logic in "View All"

### PR DESCRIPTION
**Pull request by IljaDaderko**
- Made search bar a separate component, added placeholder animation
- Simplified logic in "view all" page
   - No longer subscribing to state
   - Search state is separate from main wallet state
   - loading / search state removed from explorer controller
   - search re-uses intersection observer logic for fetching

### CODE REVIEW

File: packages/ui/src/views/w3m-wallet-explorer-view/index.ts [hunk 1]
```diff
@@ -173,49 +108,53 @@ export class W3mWalletExplorerView extends LitElement {
     else await this.onConnect(listing.desktop, listing.name)
   }
 
+  private readonly searchDebaunce = debounce((value: string) => {
```
Comment (devceline) [inline packages/ui/src/views/w3m-wallet-explorer-view/index.ts:111]:
```suggestion
  private readonly searchDebounce = debounce((value: string) => {
```

File: packages/ui/src/views/w3m-wallet-explorer-view/index.ts [hunk 2]
```diff
@@ -173,49 +108,53 @@ export class W3mWalletExplorerView extends LitElement {
...
-                : html`
-                    ${this.firstFetch
-                      ? null
-                      : html`
-                          <div class="w3m-centered-block">
-                            <w3m-text align="center" color="secondary">No results found</w3m-text>
-                          </div>
-                        `}
-                  `}
-            `}
```
Comment (devceline) [inline packages/ui/src/views/w3m-wallet-explorer-view/index.ts:218]: Sorry if I'm missing something, but don't we need this?

File: packages/ui/src/views/w3m-wallet-explorer-view/index.ts [hunk 3]
```diff
@@ -84,78 +54,43 @@ export class W3mWalletExplorerView extends LitElement {
 
   private createPaginationObserver() {
     this.intersectionObserver = new IntersectionObserver(([element]) => {
-      if (element.isIntersecting) this.fetchWalletsEndless()
+      if (element.isIntersecting && !(this.search && this.firstFetch)) this.fetchWallets()
```
Comment (devceline) [inline packages/ui/src/views/w3m-wallet-explorer-view/index.ts:57]: What if, hypothetically, a search  term resulted in a over a page of results

File: packages/ui/src/views/w3m-wallet-explorer-view/index.ts (hunk 2 above)
Comment (IljaDaderko) [reply packages/ui/src/views/w3m-wallet-explorer-view/index.ts:218]: Noup, it is handled via https://github.com/WalletConnect/web3modal/pull/597/files#diff-3f421d6903c140850f3c5abc03fb4209e31026cd27ba8ea65e3a92a6da1cfdabR56-R58

File: packages/ui/src/views/w3m-wallet-explorer-view/index.ts (hunk 3 above)
Comment (devceline) [reply packages/ui/src/views/w3m-wallet-explorer-view/index.ts:57]: Oh awesome, got it.

**Review by devceline — APPROVED** [review]
LGTM

**Pull request merged**